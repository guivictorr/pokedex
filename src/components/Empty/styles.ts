import styled, { css } from 'styled-components';

export const Container = styled.section`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
  height: 100%;

  p {
    ${({ theme }) => css`
      color: ${theme.colors.grey300};
      font-size: ${theme.fontSizes.small};
      margin-top: 10px;
    `}
  }

  a,
  button {
    max-width: 250px;
    margin-top: 40px;
  }
`;
