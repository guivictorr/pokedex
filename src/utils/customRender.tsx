import { ReactNode } from 'react';
import { render } from '@testing-library/react';
import { AnimateSharedLayout } from 'framer-motion';
import { ThemeProvider } from 'styled-components';
import GlobalStyles from '../styles/global';
import FavoritesProvider from '../context/favoriteContext';
import ModalProvider from '../context/modalContext';
import theme from '../styles/theme';

const customRender = (children: ReactNode) => {
  return render(
    <ThemeProvider theme={theme}>
      <GlobalStyles />

      <ModalProvider>
        <FavoritesProvider>
          <AnimateSharedLayout type="crossfade">{children}</AnimateSharedLayout>
        </FavoritesProvider>
      </ModalProvider>
    </ThemeProvider>,
  );
};

export default customRender;
