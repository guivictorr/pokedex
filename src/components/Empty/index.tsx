import Image from 'next/image';
import Link from '../Link';

import Button from '../Button';
import Heading from '../Heading/styles';

import { Container } from './styles';

const Empty = () => {
  return (
    <Container>
      <Image
        width={600}
        height={450}
        src="/empty.svg"
        alt="Astronauta em um espaço vazio"
      />
      <Heading level={2} fontWeight={600} fontSize="large" color="grey500">
        Está meio vazio por aqui!
      </Heading>
      <p>
        Procure por pokémons para adicioná-los aos seus <br /> favoritos.
      </p>
      <Link href="/dashboard/seeAll">
        <Button title="Procurar pokémons" outlined />
      </Link>
    </Container>
  );
};

export default Empty;
