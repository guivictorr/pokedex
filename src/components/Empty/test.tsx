import { screen } from '@testing-library/react';
import Empty from '.';
import customRender from '../../utils/customRender';

describe('<Empty/>', () => {
  it('should render correctly', () => {
    const { container } = customRender(<Empty />);

    expect(
      screen.getByRole('heading', { name: /estÃ¡ meio vazio por aqui!/i }),
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have a link to search page', () => {
    customRender(<Empty />);

    expect(screen.getByRole('link')).toHaveAttribute(
      'href',
      '/dashboard/search',
    );
  });
});
