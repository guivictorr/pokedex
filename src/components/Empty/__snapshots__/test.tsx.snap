// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Empty/> should render correctly 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  outline: transparent;
  border: none;
  border-radius: 8px;
  padding: 18px 26px;
  cursor: pointer;
  font-weight: 500;
  font-size: 16px;
  background-color: transparent;
  border: 1px solid #7d7e80;
  color: #7d7e80;
}

.c1 {
  color: #1e1e1f;
  font-weight: 600;
  font-size: 24px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  height: 100%;
}

.c0 p {
  color: #7d7e80;
  margin-top: 10px;
}

.c0 a,
.c0 button {
  max-width: 250px;
}

.c0 a {
  margin-top: 40px;
}

<section
  class="c0"
>
  <span
    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
  >
    <span
      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
    >
      <img
        alt=""
        aria-hidden="true"
        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27600%27%20height=%27450%27/%3e"
        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
      />
    </span>
    <img
      alt="Astronauta em um espaço vazio"
      data-nimg="intrinsic"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
    />
    <noscript />
  </span>
  <h2
    class="c1"
    color="gray500"
    font-size="large"
    font-weight="600"
  >
    Está meio vazio por aqui!
  </h2>
  <p>
    Procure por pokémons para adicioná-los aos seus
    <br />
     favoritos.
  </p>
  <a
    href="/dashboard/see-all"
  >
    <button
      class="c2"
    >
      Procurar pokémons
    </button>
  </a>
</section>
`;
